### PR TITLE
correctly serialize MTOM into axios data and add Content-Encoding: gzip support in soap request

### PR DESCRIPTION
**Pull request by ischisan**

### CODE REVIEW

**Comment by ischisan** [comment]
content was empty when using MTOM and Content-Lenfth was 0, so i added the logic for serializing the multiparts into the body.

**Comment by jsdevel** [comment]
please undo all the style changes

**Comment by ischisan** [comment]
> please undo all the style changes. i can't even see what you changed with all the whitespace changes...

sorry, prettier was activated ;-)

i removed the style changes.

BR

**Comment by ischisan** [comment]
I added support to enforce gzip compression in soap request

File: Readme.md [hunk 1]
```diff
@@ -522,6 +522,7 @@ Interesting properties might be:
   ]
   ```
 * `forceMTOM`: set to True if you want to send the request as MTOM even if you don't have attachments
+* `forceGzip`: set to True if you ant to force transfer-encoding in gzip (default: ```false```)
```
Comment (jsdevel) [inline Readme.md:525]: ant -> want

File: Readme.md [hunk 2]
```diff
@@ -554,6 +555,7 @@ Interesting properties might be:
...
```
Comment (jsdevel) [inline Readme.md:558]: ant -> want

**Review by jsdevel — CHANGES_REQUESTED** [review]

**Comment by jsdevel** [comment]
please ensure the build passes

**Review by jsdevel — APPROVED** [review]

**Comment by ischisan** [comment]
> @ischisan do you have a way of adding a test for this?

this is hardly possible when the soap server does not support MTOM. Maybe I can add that server functionality later, but not before next month. I would provide a test for server and client after that.

**Comment by jsdevel** [comment]
that would be great if you could follow up when you can @ischisan

**Pull request merged**